### DISCUSSION TITLE
Large SQS payloads in Lambda

### DISCUSSION THREAD

**Discussion by nilsm0hr** [aws-powertools/powertools-lambda-python#5391] [Q&A]
Hi, 
we have been playing with the large SQS payloads feature and the sqs extended client for python. While it is convenient to send SQS messages using the client, it doesn't support the Lambda use-case very well. (It expects that you use the SQS client to poll for messages, which the Lambda runtime does on my behalf https://github.com/awslabs/amazon-sqs-python-extended-client-lib?tab=readme-ov-file#enabling-support-for-payloads--256-kb)

I had a quick look at the SQS payload and the implementation, would that be something that powertools can handle? I guess it is something that could be enabled via a flag and then you just check for "software.amazon.payloadoffloading.PayloadS3Pointer" in the body and the ExtendedPayloadSize in the message attributes? 

Not sure how well this would fit with the powertools vision, but the extended client at the moment is not particularly easy to use to with AWS Lambda. 

Happy to hear your thoughts

**Comment by leandrodamascena** [comment]
Hi @nilsm0hr! Apologies for the late reply, I completely missed this discussion!

Regarding large payloads, we have an [opened issue](https://github.com/aws-powertools/powertools-lambda-python/issues/5157) and it would be good if we could start a discussion on how this should be implemented. But yes, it is something we are looking at for 2025, we see that customer demand for this is starting to increase.

Thanks for kicking out this.

**Comment by leandrodamascena** [comment]
I'm closing this discussion as resolved. Please reopen if you need.
